[DO NOT MERGE] DOCS-12770 lifecycle states (#166)
mulesoft/docs-exchange: modules/ROOT/pages/to-create-an-asset.adoc

:imagesdir: ../assets/images/
:page-aliases: to-publish-an-asset.adoc

> To share an API, API Group, policy, example, template, or connector in either an Anypoint Exchange private instance or an Exchange public portal, create an asset of that type.

Reviewer: ```suggestion
To share an API, API group, policy, example, template, or connector in either an Anypoint Exchange private instance or an Exchange public portal, create an asset of that type.
```

If the file is a ZIP, the selected main file must be in the root directory of the ZIP file. If the file is not a ZIP or if the file is a ZIP file with only one main file, then the main file is selected automatically.
+
. (Optional) Select *Advanced* and edit the advanced settings: *GroupId*, *AssetId*, *Version*, and *API version*. Exchange generates the group ID, asset ID, and version for you, and you can change these values here. You can change an API asset's version (asset version) and API version separately. The advanced settings are most often used to change the asset version.

> . Select an option for *Lifecycle state*.

Reviewer 2: I think this should be (Optional). By default, the Stable option is selected, so if nothing was clicked it would create it as a Stable asset.

> . (Optional) Select an option for *Lifecycle state*. The default is *Stable*.

Reviewer: I guess you want to maintain consistency with the peer list items, but going forward let's convert these (Optional) steps to "If you want to..." statements or "Optionally, do this...."

. Select the asset type *Custom* from the drop-down list.
. (Optional) To share a file with users, choose the file.
. (Optional) Select *Advanced* and edit the advanced settings: *GroupId*, *AssetId*, and *Version*. Exchange generates the group ID, asset ID, and version for you, and you can change these values here. The advanced settings are most often used to change the asset version.

> . (Optional) Select an option for *Lifecycle state*. The default is *Stable*.

Reviewer: ```suggestion
. Optionally select an option for *Lifecycle state*. The default is *Stable*.
```

== Create a New Version of an Existing Asset

If you have contributor or admin access to an asset, you can add a new version for an existing asset:

. In Exchange, open the asset list and select the asset.
. Click *Add version*. 
. For API assets, follow the steps for the corresponding API asset type:

> * REST API - RAML: Provide a RAML API specification file. * RAML specifications must be a RAML file (.raml).

Reviewer: ```suggestion
* REST API - RAML: Provide a RAML API specification file. A RAML specification must be a RAML file (.raml).
```

> * REST API - OAS: Provide an OAS API specification file. *OAS specifications can be either a YAML (.yaml) or JSON (.json) file. Exchange supports OAS 2.0 and OAS 3.0 specifications.

Reviewer: ```suggestion
* REST API - OAS: Provide an OAS API specification file. An OAS specification can be either a YAML (.yaml) or JSON (.json) file. Exchange supports both OAS 2.0 and OAS 3.0 specifications.
```

> * SOAP API - WSDL: Provide a WSDL API specification file. SOAP specifications file can be either a WSDL (.wsdl) or XML (.xml) file.

Reviewer: ```suggestion
* SOAP API - WSDL: Provide a WSDL API specification file. A SOAP specifications file can be either a WSDL (.wsdl) or XML (.xml) file.
```

> * API Spec Fragment - RAML : Provide an API Fragment RAML specification file. Fragment specifications must be a RAML file (.raml).

Reviewer: ```suggestion
* API Spec Fragment - RAML: Provide an API Fragment RAML specification file. A fragment specification must be a RAML file (.raml).
```

* HTTP API - This asset does not require a file. This type of asset provides an API endpoint that is defined by API Manager.
. For RAML, API Spec Fragment, OAS, and WSDL assets:

> * Select Choose File to locate the API specification file.

Reviewer: ```suggestion
* Select *Choose File* to locate the API specification file.
```

* Select the main file of the API.

> . (Optional) Choose a file to upload.

Reviewer: ```suggestion
. Optionally, choose a file to upload.
```

> . Enter the *Version* (asset version).

Reviewer: ```suggestion
. Enter a *Version* value (asset version).
```

> . (Optional) Select on option for *Lifecycle state*. The default is *Stable*.

Reviewer: ```suggestion
. Optionally select an option for *Lifecycle state*. The default is *Stable*.
```

> If you have contributor or admin access to an asset, you can add a new version from the asset portal:

Reviewer 2: There is also another way to add a version through the versions table modal (manage versions button). It may be worth adding a note about that.

. In Exchange, open the asset list and select the asset.

> . CLick *Add version*. 

Reviewer 2: ```suggestion
. Click *Add version*. 
```

> . For API assets, follow the steps for the corresponding API asst type:

Reviewer 2: ```suggestion
. For API assets, follow the steps for the corresponding API asset type:
```
